### PR TITLE
Update all samples to ASP.NET Core 3.1

### PR DESCRIPTION
**Pull request by Hawxy**
I cleaned up a lot of the custom port shenanigans so these examples should be easier to update in the future. The multi-issuer example has also been rewritten to be more ASP.NET Core idiomatic. From what I've read, I believe it retains the same behaviour as before,

### CODE REVIEW

File: Quickstart/00-Starter-Seed/Controllers/ApiController.cs [hunk 1]
```diff
@@ -8,7 +8,7 @@ public class ApiController : ControllerBase
     {
         [HttpGet]
         [Route("public")]
-        public IActionResult Public()
+        public ActionResult Public()
```
Comment (damieng) [inline Quickstart/00-Starter-Seed/Controllers/ApiController.cs]: Why was it necessary to change the IActionResult interface types to the concrete type?

File: Quickstart/00-Starter-Seed/Dockerfile [hunk 2]
```diff
@@ -1,13 +1,13 @@
...
+WORKDIR /source
 
+COPY *.csproj .
 RUN ["dotnet", "restore"]
 
+COPY . .
 RUN ["dotnet", "build"]
 
-EXPOSE 3010/tcp
+EXPOSE 5000/tcp
```
Comment (damieng) [inline Quickstart/00-Starter-Seed/Dockerfile]: Any specific reason for changing the port numbers?

File: Quickstart/00-Starter-Seed/Program.cs [hunk 3]
```diff
@@ -1,29 +1,20 @@
...
-                .UseUrls("http://*:5000")
-                .UseConfiguration(config)
-                .UseStartup<Startup>();
-        }
+        public static IHostBuilder CreateHostBuilder(string[] args) =>
+            Host.CreateDefaultBuilder(args)
+                .ConfigureWebHostDefaults(webBuilder =>
+                {
+                    webBuilder.UseStartup<Startup>();
+                });
```
Comment (damieng) [inline Quickstart/00-Starter-Seed/Program.cs]: Is this from the updated 3.1 new project template in VS?

File: Samples/multiple-issuer/Controllers/ApiController.cs [hunk 4]
```diff
@@ -20,9 +20,9 @@ public IActionResult Public()
         [HttpGet]
         [Route("private")]
         [Authorize]
-        public IActionResult Private()
+        public ActionResult Private()
         {
-            return Json(new
+            return Ok(new
```
Comment (damieng) [inline Samples/multiple-issuer/Controllers/ApiController.cs]: I think we should probably leave these as JSON.

File: Quickstart/00-Starter-Seed/Dockerfile (hunk 2 above)
Comment (Hawxy) [reply Quickstart/00-Starter-Seed/Dockerfile]: 5000/5001 is the default for a new ASP.NET Core project. It just reduces the diff between the Auth0 template and the usual out of the box experience.

File: Quickstart/00-Starter-Seed/Program.cs (hunk 3 above)
Comment (Hawxy) [reply Quickstart/00-Starter-Seed/Program.cs]: Yes.

File: Samples/multiple-issuer/Controllers/ApiController.cs (hunk 4 above)
Comment (Hawxy) [reply Samples/multiple-issuer/Controllers/ApiController.cs]: Whoops, I was testing something and forgot to change that back, will do.

File: Quickstart/00-Starter-Seed/Controllers/ApiController.cs (hunk 1 above)
Comment (Hawxy) [reply Quickstart/00-Starter-Seed/Controllers/ApiController.cs]: It wasn't really necessary. It just provides a handful of advantages over `IActionResult`, especially if you're generating swagger docs. In ASP.NET Core 2.2+ it's generally recommended over the interface, so I'm just following best practice.

File: Quickstart/00-Starter-Seed/Dockerfile (hunk 2 above)
Comment (Hawxy) [reply Quickstart/00-Starter-Seed/Dockerfile]: I've updated the docker images and returned the host port back to 3010.

File: Quickstart/00-Starter-Seed/Controllers/ApiController.cs (hunk 1 above)
Comment (damieng) [reply Quickstart/00-Starter-Seed/Controllers/ApiController.cs]: I couldn't find anything about why to use it over IActionResult - do you have a link?

Comment (Hawxy) [reply Quickstart/00-Starter-Seed/Controllers/ApiController.cs]: https://joonasw.net/view/aspnet-core-2-1-actionresult-of-t
https://docs.microsoft.com/en-us/aspnet/core/web-api/action-return-types?view=aspnetcore-3.1#actionresultt-type

It's a minor improvement, but an improvement nonetheless.

Comment (damieng) [reply Quickstart/00-Starter-Seed/Controllers/ApiController.cs]: Those examples are specifically talking all about the benefits of using `ActionResult<T>` type and the extra inference they can do knowing the `T` type for Swagger etc.

I couldn't see any benefit of just using the non-generic `ActionResult`. Given that all these methods actually return anonymous objects there isn't a T to use for the generic version.

Comment (Hawxy) [reply Quickstart/00-Starter-Seed/Controllers/ApiController.cs]: That's a valid point. I can revert that change if you'd like.

Comment (damieng) [reply Quickstart/00-Starter-Seed/Controllers/ApiController.cs]: Yeah lets go back to the IActionResult for now - the other changes look great, thanks!

**Comment by damieng** [comment]
Thanks!

**Pull request merged**